test/date-time/date-time-utility.test.ts

```diff
@@ -0,0 +1,397 @@
+        when DateTime is created from that
+        then it should have valueOf() negative (-60000)`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "1969-12-31 23:59", zone: "utc" }).valueOf(), -60000);
+            }
+        );
+
+        test(`Given a value epoch start ("1970-01-01 00:01") in utc
+        when DateTime is created from that
+        then it should have valueOf() positive (60000)`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "1970-01-01 00:01", zone: "utc" }).valueOf(), 60000);
+            }
+        );
+    });
+
+
+    suite("to string", () =>
+    {
+        const value = "2024-01-01 10:00";
+
+        test(`Given a valid value (${value}) and zone (utc)
+        when a DateTime is created from that value and zone
+        then toString() on that dateTime should return a string with value and zone`,
+            () =>
+            {
+                const zone = "utc";
+                const dateTime = new DateTime({ value: value, zone: zone });
+                Assert.strictEqual(dateTime.toString(), `${value} ${zone}`);
+            }
+        );
+
+        test(`Given a valid value (${value}) and zone (UTC+5:30)
+        when a DateTime is created from that value and zone
+        then toString() on that dateTime should return a string with value and zone`,
+            () =>
+            {
+                const zone = "UTC+5:30";
+                const dateTime = new DateTime({ value: value, zone: zone });
+                Assert.strictEqual(dateTime.toString(), `${value} ${zone}`);
+            }
+        );
+
+        test(`Given a valid value (${value}) and zone (America/Los_Angeles)
+        when a DateTime is created from that value and zone
+        then toString() on that dateTime should return a string with value and zone`,
+            () =>
+            {
+                const zone = "America/Los_Angeles";
+                const dateTime = new DateTime({ value: value, zone: zone });
+                Assert.strictEqual(dateTime.toString(), `${value} ${zone}`);
+            }
+        );
+    });
+
+    suite("to string date time", () =>
+    {
+        const value = "2024-01-01 10:00";
+
+        test(`Given a valid value (${value}) and zone (utc)
+        when a DateTime is created from that value and zone
+        then toStringDateTime() on that dateTime should return the passed in value`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value, zone: "utc" }).toStringDateTime(), value);
+            }
+        );
+
+        test(`Given a valid value (${value}) and zone (utc)
+        when a DateTime is created from that value and zone
+        then toStringDateTime() should return the value in a valid DateTime format`,
+            () =>
+            {
+                Assert.ok(DateTime.validateDateTimeFormat(new DateTime({ value: value, zone: "utc" }).toStringDateTime()));
```

Comment:
?

Reply:
for toString have a few examples where you know the output and just check against it